File: Searching/LinearSearch.cpp
#include <iostream>

using namespace std;

int linearSearch(int arr[], int target, int n) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == target) {
            return i;
        }
    }
    return -1; // Return -1 if the target element is not found
}

int main() {
    int arr[] = {10, 20, 30, 40, 50};
    int target = 40;
    int n = 5;

    int index = linearSearch(arr, target, n);
    if (index != -1) {
        cout << "Element found at index: " << index << endl;
    } else {
        cout << "Element not found" << endl;
    }
    
    return 0;
}

File: arrays/LinearSearch.java
public class LinearSearch {
    
    static boolean linearSearch(int[] arr, int size, int target) {
        for (int i = 0; i < size; i++) {
            if (arr[i] == target) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        int arr[] = {2, 4, 6, 8, 10};
        int size = 5;
        int target = 10;

        boolean ans = linearSearch(arr, size, target);
        if (ans) {
            System.out.println("Target found");
        } else {
            System.out.println("Target not found");
        }
    }
}
